lib/services/plugins-service.ts

```diff
@@ -43,14 +59,64 @@ export class PluginsService implements IPluginsService {
 
 	public remove(pluginName: string): IFuture<void> {
 		return (() => {
+			let isUninstallCommandExecuted = false;
+
+			let executeUninstallCommand = () => {
+				return (() => {
+					if(!isUninstallCommandExecuted) {
+						this.executeNpmCommand(PluginsService.UNINSTALL_COMMAND_NAME, pluginName).wait();
+						isUninstallCommandExecuted = true;
+					}
+				}).future<void>()();
+			};
+
 			let removePluginNativeCodeAction = (modulesDestinationPath: string, platform: string, platformData: IPlatformData) => {
-				let pluginData = this.convertToPluginData(this.getNodeModuleData(pluginName).wait());
-				pluginData.isPlugin = true;
-				return platformData.platformProjectService.removePluginNativeCode(pluginData);
+				return (() => {
+					let pluginData = this.convertToPluginData(this.getNodeModuleData(pluginName).wait());
+
+					platformData.platformProjectService.removePluginNativeCode(pluginData).wait();
+
+					// Remove the plugin and call merge for another plugins that have configuration file
+					let pluginConfigurationFilePath = this.getPluginConfigurationFilePath(pluginData, platformData);
+					if(this.$fs.exists(pluginConfigurationFilePath).wait()) {
+						let tnsModulesDestinationPath = path.join(platformData.appDestinationDirectoryPath, constants.APP_FOLDER_NAME, constants.TNS_MODULES_FOLDER_NAME);
+						let nodeModules = this.$broccoliBuilder.getChangedNodeModules(tnsModulesDestinationPath, platform).wait();
+						let plugins = _.map(_.keys(nodeModules), nodeModule => this.convertToPluginData(this.getNodeModuleData(pluginName).wait()));
+
+						_.forEach(plugins, (data, index) => {
+							if(data.isPlugin) {
+								if(this.$fs.exists(this.getPluginConfigurationFilePath(data, platformData)).wait()) {
+									executeUninstallCommand().wait();
+
+									if(index === 0) {
+										this.$projectDataService.initialize(this.$projectData.projectDir);
```

Comment:
I think the entire code  in the body of the if should be moved into a dedicated function - `initializeFromCache` (name could be improved, of course)